api/test/e2e/route_test.go

```diff
@@ -0,0 +1,231 @@
+            }
+        }`).
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusBadRequest)
+
+	//create route  -- invalid host
+	MangerApiExpect(t).
+		PUT("/apisix/admin/routes/r2").
+		WithText(`{
+            "uri": "/hello_",
+            "host": "$%$foo.com",
+            "upstream": {
+                "nodes": {
+                    "172.16.238.120:1980": 1
+                },
+                "type": "roundrobin"
+            }
+        }`).
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusBadRequest)
+
+		//create route  -- invalid type for host
+	MangerApiExpect(t).
+		PUT("/apisix/admin/routes/r2").
+		WithText(`{
+            "uri": "/hello_",
+            "host": 1,
+            "upstream": {
+                "nodes": {
+                    "172.16.238.120:1980": 1
+                },
+                "type": "roundrobin"
+            }
+        }`).
+		WithHeader("Authorization", token).
+		Expect().
+		//Status(http.StatusBadRequest)
+		JSON().Object().ValueNotEqual("code", 0)
+
+	//create route use host
+	MangerApiExpect(t).PUT("/apisix/admin/routes/r2").WithText(`{
+            "uri": "/hello_",
+            "host": "test.com",
+            "upstream": {
+                "nodes": {
+                    "172.16.238.120:1980": 1
+                },
+                "type": "roundrobin"
+            }
+        }`).
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusOK)
+
+	//sleep
+	time.Sleep(time.Duration(100) * time.Millisecond)
+
+	//hit route - ok
+	APISIXExpect(t).GET("/hello_").
+		WithHeader("Host", "test.com").
+		Expect().
+		Status(http.StatusOK)
+
+	//create route without host and hosts
+	MangerApiExpect(t).
+		PUT("/apisix/admin/routes/r3").
+		WithText(`{
+            "uri": "/hello1",
+            "upstream": {
+                "nodes": {
+                    "172.16.238.120:1980": 1
+                },
+                "type": "roundrobin"
+            }
+        }`).
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusOK)
+
+	//sleep
+	time.Sleep(time.Duration(100) * time.Millisecond)
+
+	//hit route - ok, match any host
+	APISIXExpect(t).GET("/hello1").
+		WithHeader("Host", "test.com").
+		Expect().
+		Status(http.StatusOK)
+
+	//hit route - ok, match any host
+	APISIXExpect(t).GET("/hello1").
+		WithHeader("Host", "foo.com").
+		Expect().
+		Status(http.StatusOK)
+
+	//hit route -- not found, wrong uri
+	APISIXExpect(t).GET("/not_found").
+		Expect().
+		Status(http.StatusNotFound)
+
+	//delete test data
+	MangerApiExpect(t).
+		DELETE("/apisix/admin/routes/r3").
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusOK)
+
+	MangerApiExpect(t).
+		DELETE("/apisix/admin/routes/r2").
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusOK)
+
+	MangerApiExpect(t).
+		DELETE("/apisix/admin/routes/r1").
+		WithHeader("Authorization", token).
+		Expect().
+		Status(http.StatusOK)
+
```

Comment:
Need to add test cases to confirm whether the data plane can no longer be routed after deleting the route

Reply:
updated.